cmd/seth.go

```diff
@@ -28,14 +26,54 @@ func RunCLI(args []string) error {
 			&cli.StringFlag{Name: "networkName", Aliases: []string{"n"}},
 		},
 		Before: func(cCtx *cli.Context) error {
-			var err error
 			networkName := cCtx.String("networkName")
 			if networkName == "" {
 				return errors.New(ErrNoNetwork)
 			}
 			_ = os.Setenv("NETWORK", networkName)
 			if cCtx.Args().Len() > 0 && cCtx.Args().First() != "trace" {
-				C, err = seth.NewClient()
+				var err error
+				switch cCtx.Args().First() {
+				case "keys":
+					var cfg *seth.Config
+					cfg, err = seth.ReadConfig()
+					if err != nil {
+						return err
+					}
+					keyfilePath := os.Getenv(seth.KEYFILE_PATH_ENV_VAR)
+					if keyfilePath == "" {
+						return errors.New("No keyfile path specified in KEYFILE_PATH_ENV_VAR env var")
+					}
+					cfg.KeyFileSource = seth.KeyFileSourceFile
+					cfg.KeyFilePath = keyfilePath
+					C, err = seth.NewClientWithConfig(cfg)
+					if err != nil {
+						return err
+					}
+				case "gas":
+					var cfg *seth.Config
+					var pk string
+					_, pk, err = seth.NewAddress()
```

Comment:
We'll clean up this later as agreed. Need to refactor key handling and client struct